manager/apis/app/v1alpha1/plotter_types.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PlotterSpec defines the desired state of Plotter, which is applied in a multi-clustered environment. Plotter installs the runtime environment
+// (as blueprints running on remote clusters) which provides the Data Scientist's application with secure and governed access to the data requested in the
+// M4DApplication.
+type PlotterSpec struct {
+
+	// Selector enables to connect the resource to the application
+	// Should match the selector of the owner - M4DApplication CRD.
+	Selector metav1.LabelSelector `json:"selector"`
+
+	// +required
+	// +kubebuilder::validation:MinItems:=1
+	// Blueprints structure represents remote blueprints mapped by the identifier of a cluster in which they will be running
+	Blueprints map[string]BlueprintSpec `json:"blueprints"`
+}
+
+// PlotterStatus defines the observed state of Plotter
+// This includes readiness, error message, and indicators received from blueprint
+// resources owned by the Plotter for cleanup and status monitoring
+type PlotterStatus struct {
```

Comment:
The status structs of M4DApplication, Plotter and Blueprint are almost the same.  Would it make sense to have a common structure or sub-structure?

Reply:
Created a common structure for Plotter and Blueprint states. M4DApplication has a different structure and requires some pre-processing in status update.